src/ORM/Locator/TableLocator.php

```diff
@@ -78,6 +94,33 @@ public function __construct(?array $locations = null)
         }
     }
 
+    /**
+     * Set if fallback class should be used.
+     *
+     * @param bool $enable Flag to enable or disable fallback
+     * @return $this
+     */
+    public function useFallbackClass(bool $enable)
```

Comment:
Are you thinking that this should become the default mode? How would enabling this globally work with Bake which also relies on auto-models extensively.

Reply:
Eventually yes but I would like to hear opinions on whether we should do it 4.2. We could keep the default behavior same in core and instead add the code I showed in app skeleton for 4.2 also.

In either case we can check for CLI and turn on fallback mode in `Application::bootstrap()` or in bake's `Plugin::bootstrap()` or `BakeCommand`'s constructor.

Reply:
> Eventually yes but I would like to hear opinions on whether we should do it for 4.2. We could keep the default behavior same in core and instead add the code I showed in app skeleton for 4.2 also.

Ok. I agree with your earlier statement that auto-tables are a boon and a curse. They are great for getting started and prototyping but quickly become a liability.

I like your idea of enabling auto-tables in bake and the console command to smooth over the experience for new applications.

Reply:
I'd like to see this rather sooner than later, and if it basically only affects newly create applications, then it shouldn't really be a problem. The only problem might be people upgrading their bootstrap according to `cakephp/app` when upgrading to 4.2, but I would assume that the people that do that are usually the more savvy ones.

Reply:
We hope to have 4.2 out "soon" to make this feature available :slightly_smiling_face: